app/src/main/java/com/example/nastya/laba/controller/Controller.java

```diff
@@ -0,0 +1,56 @@
+package com.example.nastya.laba.controller;
+
+import com.example.nastya.laba.interfaces.RedditApi;
+import com.example.nastya.laba.model.Feed;
+import com.example.nastya.laba.model.children.Children;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class Controller implements Callback <Feed> {
+
+    private final static String BASE_URL = "https://reddit.com/";
+    private final static String TAG = "Controller";
+
+
+    public void start() {
+        Retrofit retrofit = new Retrofit.Builder()
```

Comment:
This is not a completely valid place to create your retrofit client. Right approach is to create your retrofit client along with API client in `Application` and then access it from Controller using `((ApplicationEx) getApplication()).getApiClient()`. 

This will become important when you will have two screens, that need to use same API client